### PR TITLE
add gradle tasks to publish artifacts

### LINKED ISSUE #11: Setup gradle tasks to publish artifacts
We need to be able to cut releases of the connector in the same way that we do with flink connectors. See https://github.com/pravega/flink-connectors/blob/master/gradle/maven.gradle

### PR DESCRIPTION
**Pull request by yangb8**
Closes #11.

Generated files: 
```
$ gradle uploadShadow -PpublishUrl=file:///home/bo/go/src/github.com/yangb8/hadoop-connectors/local -PdoSigning
...
BUILD SUCCESSFUL in 3s
9 actionable tasks: 1 executed, 8 up-to-date

$ ll local/io/pravega/hadoop-connectors/0.3.0-SNAPSHOT/
total 19212
drwxrwxr-x 2 bo bo     4096 Feb 24 11:52 ./
drwxrwxr-x 3 bo bo     4096 Feb 24 11:52 ../
-rw-rw-r-- 1 bo bo 19499491 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.jar
-rw-rw-r-- 1 bo bo      475 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.jar.asc
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.jar.asc.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.jar.asc.sha1
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.jar.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.jar.sha1
-rw-rw-r-- 1 bo bo    51925 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-javadoc.jar
-rw-rw-r-- 1 bo bo      475 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-javadoc.jar.asc
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-javadoc.jar.asc.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-javadoc.jar.asc.sha1
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-javadoc.jar.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-javadoc.jar.sha1
-rw-rw-r-- 1 bo bo     1098 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.pom
-rw-rw-r-- 1 bo bo      475 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.pom.asc
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.pom.asc.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.pom.asc.sha1
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.pom.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1.pom.sha1
-rw-rw-r-- 1 bo bo     9812 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-sources.jar
-rw-rw-r-- 1 bo bo      475 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-sources.jar.asc
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-sources.jar.asc.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-sources.jar.asc.sha1
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-sources.jar.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 hadoop-connectors-0.3.0-20180224.165229-1-sources.jar.sha1
-rw-rw-r-- 1 bo bo      363 Feb 24 11:52 maven-metadata.xml
-rw-rw-r-- 1 bo bo       32 Feb 24 11:52 maven-metadata.xml.md5
-rw-rw-r-- 1 bo bo       40 Feb 24 11:52 maven-metadata.xml.sha1
```

### CODE REVIEW

**Comment by EronWright** [comment]
I'm not thrilled by the amount of custom build logic is needed.   Can you add some commentary and/or simplify?

How consistent is this with the Flink connector?

**Comment by yangb8** [comment]
@EronWright 

The logic of signing and publication is almost identical as the one in Flink connector. And could you please elaborate a little bit regarding the custom build logic?

File: gradle/maven.gradle [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                connection 'scm:git:git://github.com/pravega/hadoop-connectors.git'
+                developerConnection 'scm:git:https://github.com/pravega/hadoop-connectors.git'
+            }
+            licenses {
+                license {
+                    name 'The Apache License, Version 2.0'
+                    url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
+                }
+            }
+            developers {
```
Comment (EronWright) [inline gradle/maven.gradle]: Let's not list individual developers.

**Comment by EronWright** [comment]
By custom build logic, I mean the new tasks in `java.gradle`.

File: README.md [hunk 2]
```diff
@@ -58,11 +58,11 @@ Run Examples
 ```
 Hadoop (verified with Hadoop 2.8.1 on Ubuntu 16.04)
```
Comment (EronWright) [inline README.md]: I notice on PR #9 that Hadoop 3.0 was used.   Can we stick to 2.8 for now?

File: gradle/maven.gradle (hunk 1 above)
Comment (yangb8) [reply gradle/maven.gradle]: accepted

File: README.md (hunk 2 above)
Comment (yangb8) [reply README.md]: yes, will revert it to 2.8.3

**Review by EronWright — APPROVED** [review]
LGTM

**Pull request merged**